jupyter_server_nbmodel/handlers.py

```diff
@@ -219,15 +228,30 @@ async def _execute_snippet(
         The execution status and outputs.
     """
     ycell = None
+    time_info = {}
     if metadata is not None:
         ycell = await _get_ycell(ydoc, metadata)
         if ycell is not None:
+            execution_start_time = datetime.now(timezone.utc).isoformat()[:-6]
             # Reset cell
             with ycell.doc.transaction():
                 del ycell["outputs"][:]
                 ycell["execution_count"] = None
                 ycell["execution_state"] = "running"
-
+                if metadata.get("record_timing", False):
+                    time_info = ycell["metadata"].get("execution", {})
```

Comment:
We need to clear `ycell["metadata"]["execution"]` if it exists. Otherwise, old timing for `shell.execute_reply` and `execution_failed` is stored when re-executing the cell. In general we should clear it at the same time when we schedule cell execution (see https://github.com/datalayer/jupyter-server-nbmodel/issues/30), but for the sake of this PR we can clear it here.